src/portals/hope/features/claims/claim-details/ClaimCounterparty/CreateCounterpartyModal.tsx

```diff
@@ -0,0 +1,127 @@
+import styled from '@emotion/styled'
+import {
+  convertEnumToTitle,
+  Dropdown,
+  DropdownOption,
+  Input,
+  isPressing,
+  Keys,
+  Modal,
+  Paragraph,
+  Shadowed,
+  ThirdLevelHeadline,
+  usePlatform,
+} from '@hedvig-ui'
+import gql from 'graphql-tag'
+import React, { useState } from 'react'
+import { toast } from 'react-hot-toast'
+import { useAddClaimCounterpartyMutation } from 'types/generated/graphql'
+
+gql`
+  mutation AddClaimCounterparty($claimId: ID!, $reference: String!) {
+    claim_addCounterparty(claimId: $claimId, reference: $reference) {
+      id
+      counterparties {
+        ...ClaimCounterpartyInfo
+      }
+    }
+  }
+`
+
+const CounterpartyModal = styled(Modal)`
+  width: 23rem;
+  padding: 15px;
+  background-color: ${({ theme }) => theme.backgroundLight};
+`
+
+enum CounterpartyType {
+  Car = 'CAR',
+  Other = 'OTHER',
+}
+
+const SubmitTip = styled(Paragraph)`
+  text-align: right;
+  font-size: 0.8em;
+  margin-top: 1.25rem;
+  color: ${({ theme }) => theme.semiStrongForeground};
+  transition: opacity 0.5s;
+`
+
+export const CreateCounterpartyModal: React.FC<{
+  visible: boolean
+  onClose: () => void
+  claimId: string
+}> = ({ visible, onClose, claimId }) => {
+  const { isMetaKey, metaKey } = usePlatform()
+
+  const [addClaimCounterparty] = useAddClaimCounterpartyMutation()
+
+  const [reference, setReference] = useState<string>('')
+
+  const [counterpartyType, setCounterpartyType] = useState<CounterpartyType>(
+    CounterpartyType.Car,
+  )
+
+  const addCounterpartyHandler = async () => {
+    await toast.promise(
+      addClaimCounterparty({
+        variables: {
+          claimId,
+          reference,
+        },
+      }),
+      {
+        loading: 'Creating counterparty...',
+        success: 'Counterparty successfully created!',
+        error: 'Error while creating counterparty!',
+      },
+    )
+    onClose()
+  }
+
+  return (
+    <CounterpartyModal visible={visible} onClose={onClose}>
+      <ThirdLevelHeadline>Add new counterparty</ThirdLevelHeadline>
+      <Dropdown placeholder="Type" style={{ marginBottom: '1rem' }}>
+        {Object.values(CounterpartyType).map((type) => (
+          <DropdownOption
+            key={type}
+            selected={counterpartyType === type}
+            onClick={() => setCounterpartyType(type)}
+          >
+            {convertEnumToTitle(type)}
+          </DropdownOption>
+        ))}
+      </Dropdown>
+      <Input
+        label="External reference"
+        placeholder={
+          counterpartyType === CounterpartyType.Car
+            ? 'Registration number'
+            : 'Company name, etc...'
+        }
+        value={reference}
+        onChange={(e) => {
```

Comment:
I know I wrote this but 🤪 
```suggestion
        onChange={({currentTarget: {value}}) => {
```